pkg/registry/common/connect/server_ns_test.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/networkservicemesh/sdk/pkg/registry/memory"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+	"testing"
+)
+
+func localNSRegistryServer(proxyDomain string, resolver dnsresolve.Resolver) registry.NetworkServiceRegistryServer {
+	return next.NewNetworkServiceRegistryServer(
+		memory.NewNetworkServiceRegistryServer(),
+		dnsresolve.NewNetworkServiceRegistryServer(
+			dnsresolve.WithProxyDomain(proxyDomain),
+			dnsresolve.WithResolver(resolver)),
+		connect.NewNetworkServiceRegistryServer(),
+	)
+}
+
+func proxyNSRegistryServer(domain, floatingDomain string, resolver dnsresolve.Resolver) registry.NetworkServiceRegistryServer {
+	return next.NewNetworkServiceRegistryServer(
+		dnsresolve.NewNetworkServiceRegistryServer(
+			dnsresolve.WithProxyDomain(floatingDomain),
+			dnsresolve.WithResolver(resolver)),
+		swap.NewNetworkServiceServer(domain),
+		connect.NewNetworkServiceRegistryServer(),
+	)
+}
+
+/*
+TestInterdomainNetworkServiceRegistry covers the next scenario:
+	1. local registry from domain2 has entry "ns-1"
+	2. nsmgr from domain1 call find with query "ns-1@domain2"
+	3. local registry proxies query to proxy registry
+	4. proxy registry proxies query to local registry from domain2
+
+Expected: nsmgr found ns
+
+
+domain1                                      domain2
+ ___________________________________         ___________________
+|                                   | Find  |                   |
+| local registry --> proxy registry | ----> | local registry    |
+|                                   |       |                   |
+____________________________________     	 ___________________
+*/
+func TestInterdomainNetworkServiceRegistry(t *testing.T) {
+	defer goleak.VerifyNone(t)
+	tool := newInterdomainTestingTool(t)
+	defer tool.cleanup()
+
+	const localRegistryDomain = "domain1.local.registry"
+	const proxyRegistryDomain = "domain1.proxy.registry"
+	const remoteRegistryDomain = "domain2.local.registry"
+
+	tool.startNetworkServiceRegistryServerAsync(localRegistryDomain, localNSRegistryServer(proxyRegistryDomain, tool))
+	tool.startNetworkServiceRegistryServerAsync(proxyRegistryDomain, proxyNSRegistryServer(proxyRegistryDomain, "", tool))
+
+	remoteMem := memory.NewNetworkServiceRegistryServer()
+	_, err := remoteMem.Register(context.Background(), &registry.NetworkService{Name: "ns-1"})
+	require.Nil(t, err)
+
+	tool.startNetworkServiceRegistryServerAsync(remoteRegistryDomain, remoteMem)
+
+	client := registry.NewNetworkServiceRegistryClient(tool.dialDomain(localRegistryDomain))
+
+	stream, err := client.Find(context.Background(), &registry.NetworkServiceQuery{
+		NetworkService: &registry.NetworkService{
+			Name: "ns-1@" + remoteRegistryDomain,
+		},
+	})
+
+	require.Nil(t, err)
+
+	list := registry.ReadNetworkServiceList(stream)
+
+	require.Len(t, list, 1)
+	require.Equal(t, "ns-1@"+remoteRegistryDomain, list[0].Name)
+}
+
+/*
+TestInterdomainFloatingNetworkServiceRegistry covers the next scenario:
```

Comment:
@edwarnicke can you please check this scenario?

Reply:
Full file link https://github.com/networkservicemesh/sdk/blob/c65d0fce8ee6e9e0c212846e62611f49727d7ccf/pkg/registry/common/connect/server_ns_test.go